### DISCUSSION TITLE
[bevy reflect] nest struct or other nest types could be applied recursively?

### DISCUSSION THREAD

**Discussion by greenhat616** [bevyengine/bevy#15466] [Q&A]
Hello. I have checked the document and examples. And I haven't noticed a tips on nested type.

Here is a example type defination below. And is it possile create a nest `DynamicStruct` or just use the same logic in `GetPath`(via `.` access nest struct) to patch the nest object?

Thank you!

```rust
#[derive(Reflect)]
struct NestedType {
  pub a: usize;
  pub b: usize
}

#[derive(Reflect)]
struct Parent {
  nested: NestedType
}
```

**Comment by MrGVSV** [comment]
If you’re asking whether nesting is possible: yes it is. The example you gave should compile fine (excluding the semicolon typo).

You can access the nested fields through `GetPath` like you mentioned, or through `Struct::field` + `Reflect::reflect_ref`.

**Reply by greenhat616** [reply]
If I understand correctly, there are two possible ways to "patch" nest type with a `DynamicStruct`, right?

```rust
// Option 1: Use nested DynamicStruct
let mut dynamic_struct = DynamicStruct::default();
let mut nested_struct = DynamicStruct::default();
nested_struct.insert("a", 42u32);
dynamic_struct.insert("nested", nested_struct);
parent.apply(&dynamic_struct);

// Option 2: Use `GetPath`
let mut dynamic_struct = DynamicStruct::default();
dynamic_struct.insert("nested.a", 42u32);
parent.apply(&dynamic_struct);
```

Would this approach work as expected?

Thank you!
